x/projects/keeper/creation.go

```diff
@@ -2,61 +2,139 @@ package keeper
 
 import (
 	"math"
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/lavanet/lava/utils"
+	plantypes "github.com/lavanet/lava/x/plans/types"
 	"github.com/lavanet/lava/x/projects/types"
 )
 
 // add a default project to a subscription, add the subscription key as
-func (k Keeper) CreateAdminProject(ctx sdk.Context, subscriptionAddress string, totalCU uint64, cuPerEpoch uint64, providers uint64, vrfpk string) error {
-	return k.CreateProject(ctx, subscriptionAddress, types.ADMIN_PROJECT_NAME, subscriptionAddress, true, totalCU, cuPerEpoch, providers, math.MaxUint64, vrfpk)
+func (k Keeper) CreateAdminProject(ctx sdk.Context, subscriptionAddress string, plan plantypes.Plan, vrfpk string) error {
+	projectData := types.ProjectData{
+		Name:        types.ADMIN_PROJECT_NAME,
+		Description: types.ADMIN_PROJECT_DESCRIPTION,
+		Enabled:     true,
+		ProjectKeys: []types.ProjectKey{{Key: subscriptionAddress, Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_DEVELOPER}, Vrfpk: vrfpk}},
+		Policy:      types.Policy{},
+	}
+	return k.CreateProject(ctx, subscriptionAddress, projectData, plan)
 }
 
 // add a new project to the subscription
-func (k Keeper) CreateProject(ctx sdk.Context, subscriptionAddress string, projectName string, adminAddress string, enable bool, totalCU uint64, cuPerEpoch uint64, providers uint64, geolocation uint64, vrfpk string) error {
-	project := types.CreateProject(subscriptionAddress, projectName)
-	var emptyProject types.Project
+func (k Keeper) CreateProject(ctx sdk.Context, subscriptionAddress string, projectData types.ProjectData, plan plantypes.Plan) error {
+	if len(projectData.GetName()) > types.MAX_PROJECT_NAME_LEN || len(projectData.GetDescription()) > types.MAX_PROJECT_DESCRIPTION_LEN {
```

Comment:
lets move this to the already existing:
func validateProjectName(projectName string) bool